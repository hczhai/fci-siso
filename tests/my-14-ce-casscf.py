import os
import sys

sys.path[:0] = [os.path.join(os.path.dirname(
    os.path.abspath(__file__)), "..")]

from pyscf import gto, scf, mcscf
from fcisiso import FCISISO
import numpy as np

mol = gto.M(atom='Ce 0 0 0',
            symmetry=False,
            basis='ccpvtz-dk', spin=1, charge=3, verbose=3)
mf = scf.newton(scf.RHF(mol).sfx2c1e())
mf.kernel()

print("\n Molecular orbital components:\n")

# total: 55 elec
# core: 1s2 2s2 2p6 3s2 3p6 3d10 4s2 4p6 4d10 5s2 5p6 (54e, 27o)
# active: 4f1 (1e, 7o)
ao_lb = gto.sph_labels(mol)
for i in range(40):
    ivs = np.argsort(mf.mo_coeff[:, i]**2)
    ss = ""
    for iv in ivs[::-1][:3]:
        ss += "%10.6f (%s)" % (mf.mo_coeff[iv, i] ** 2, ao_lb[iv].strip())
    print(" MO %2d :: %s" % (i, ss))

print("\n CASSCF(1, 7):\n")

cas_list = [28, 29, 30, 31, 32, 33, 34]
mc = mcscf.CASSCF(mf, 7, 1)
mc.fix_spin_(ss=0.75)
mo = mcscf.sort_mo(mc, mf.mo_coeff, cas_list, base=0)
mc.state_average_((1/7, 1/7, 1/7, 1/7, 1/7, 1/7, 1/7))
mc.kernel(mo)

print(mc.nelecas, mc.e_states)

dmao = mc.make_rdm1()
print(dmao.shape)

print("\n New active orbitals:\n")

for i in range(20, 40):
    ivs = np.argsort(mc.mo_coeff[:, i]**2)
    ss = ""
    for iv in ivs[::-1][:3]:
        ss += "%10.6f (%s)" % (mc.mo_coeff[iv, i] ** 2, ao_lb[iv].strip())
    print(" MO %2d :: %s" % (i, ss))

states = [(7, 2)]
ci = [
    (*mc.nelecas[::-1], 1, -1, mc.e_states[0], mc.ci[0].T),
    (*mc.nelecas[::-1], 1, -1, mc.e_states[1], mc.ci[1].T),
    (*mc.nelecas[::-1], 1, -1, mc.e_states[2], mc.ci[2].T),
    (*mc.nelecas[::-1], 1, -1, mc.e_states[3], mc.ci[3].T),
    (*mc.nelecas[::-1], 1, -1, mc.e_states[4], mc.ci[4].T),
    (*mc.nelecas[::-1], 1, -1, mc.e_states[5], mc.ci[5].T),
    (*mc.nelecas[::-1], 1, -1, mc.e_states[6], mc.ci[6].T),
    (*mc.nelecas, 1, 1, mc.e_states[0], mc.ci[0]),
    (*mc.nelecas, 1, 1, mc.e_states[1], mc.ci[1]),
    (*mc.nelecas, 1, 1, mc.e_states[2], mc.ci[2]),
    (*mc.nelecas, 1, 1, mc.e_states[3], mc.ci[3]),
    (*mc.nelecas, 1, 1, mc.e_states[4], mc.ci[4]),
    (*mc.nelecas, 1, 1, mc.e_states[5], mc.ci[5]),
    (*mc.nelecas, 1, 1, mc.e_states[6], mc.ci[6]),
]

siso = FCISISO(mol, mc, states)
siso.ci = ci
siso.ncore = 27
siso.norb = 7
energies = siso.kernel(dmao=dmao, amfi=True)

print(energies)

e0 = np.average(energies[0:6])
e1 = np.average(energies[6:14])

au2ev = 27.21139
print("")
print("E 2F(5/2)         = %10.4f eV" % (e0 * au2ev))
print("E 2F(7/2)         = %10.4f eV" % (e1 * au2ev))
print("2F(7/2) - 2F(5/2) = %10.4f eV" % ((e1 - e0) * au2ev))
print("2F(7/2) - 2F(5/2) = %10.4f cm-1" % ((e1 - e0) * au2ev * 8065.56))
