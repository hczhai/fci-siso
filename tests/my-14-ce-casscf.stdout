
WARN: HOMO -0.7946901445743663 > LUMO -0.8710833560540445 was found in the canonicalized orbitals.

converged SCF energy = -8854.72197243344

 Molecular orbital components:

 MO  0 ::   0.967582 (0 Ce 1s)  0.032329 (0 Ce 2s)  0.000088 (0 Ce 3s)
 MO  1 ::   0.899777 (0 Ce 2s)  0.069179 (0 Ce 3s)  0.030940 (0 Ce 1s)
 MO  2 ::   0.779367 (0 Ce 2py)  0.142556 (0 Ce 2px)  0.041694 (0 Ce 3py)
 MO  3 ::   0.805704 (0 Ce 2px)  0.142100 (0 Ce 2py)  0.043103 (0 Ce 3px)
 MO  4 ::   0.920549 (0 Ce 2pz)  0.049247 (0 Ce 3pz)  0.027710 (0 Ce 2py)
 MO  5 ::   0.927557 (0 Ce 3s)  0.067874 (0 Ce 2s)  0.003069 (0 Ce 4s)
 MO  6 ::   0.637764 (0 Ce 3py)  0.280427 (0 Ce 3px)  0.034203 (0 Ce 2py)
 MO  7 ::   0.655986 (0 Ce 3px)  0.291371 (0 Ce 3py)  0.035181 (0 Ce 2px)
 MO  8 ::   0.918031 (0 Ce 3pz)  0.049234 (0 Ce 2pz)  0.018383 (0 Ce 3py)
 MO  9 ::   0.921831 (0 Ce 3dxy)  0.031955 (0 Ce 3dxz)  0.024618 (0 Ce 3dz^2)
 MO 10 ::   0.934593 (0 Ce 3dx2-y2)  0.040106 (0 Ce 3dyz)  0.024592 (0 Ce 3dxy)
 MO 11 ::   0.635288 (0 Ce 3dyz)  0.280073 (0 Ce 3dxz)  0.048638 (0 Ce 3dz^2)
 MO 12 ::   0.673179 (0 Ce 3dxz)  0.230590 (0 Ce 3dyz)  0.059795 (0 Ce 3dz^2)
 MO 13 ::   0.866424 (0 Ce 3dz^2)  0.092752 (0 Ce 3dyz)  0.015735 (0 Ce 3dxy)
 MO 14 ::   0.985358 (0 Ce 4s)  0.010723 (0 Ce 5s)  0.003162 (0 Ce 3s)
 MO 15 ::   0.686398 (0 Ce 4py)  0.295051 (0 Ce 4px)  0.010324 (0 Ce 4pz)
 MO 16 ::   0.689449 (0 Ce 4px)  0.274511 (0 Ce 4py)  0.027814 (0 Ce 4pz)
 MO 17 ::   0.953635 (0 Ce 4pz)  0.030864 (0 Ce 4py)  0.007274 (0 Ce 4px)
 MO 18 ::   0.659348 (0 Ce 4dxy)  0.275345 (0 Ce 4dx2-y2)  0.029971 (0 Ce 4dxz)
 MO 19 ::   0.657542 (0 Ce 4dx2-y2)  0.293681 (0 Ce 4dxy)  0.044939 (0 Ce 4dyz)
 MO 20 ::   0.581021 (0 Ce 4dyz)  0.326363 (0 Ce 4dxz)  0.056525 (0 Ce 4dx2-y2)
 MO 21 ::   0.628278 (0 Ce 4dxz)  0.278962 (0 Ce 4dyz)  0.061087 (0 Ce 4dz^2)
 MO 22 ::   0.880158 (0 Ce 4dz^2)  0.080831 (0 Ce 4dyz)  0.015918 (0 Ce 4dxy)
 MO 23 ::   0.859161 (0 Ce 5s)  0.128602 (0 Ce 6s)  0.010810 (0 Ce 4s)
 MO 24 ::   0.950220 (0 Ce 5pz)  0.033601 (0 Ce 5py)  0.018889 (0 Ce 7pz)
 MO 25 ::   0.735256 (0 Ce 5px)  0.233397 (0 Ce 5py)  0.017313 (0 Ce 5pz)
 MO 26 ::   0.718968 (0 Ce 5py)  0.248566 (0 Ce 5px)  0.018432 (0 Ce 5pz)
 MO 27 ::   0.449094 (0 Ce 6s)  0.282890 (0 Ce 7s)  0.057377 (0 Ce 5s)
 MO 28 ::   0.851701 (0 Ce 4f+1)  0.076468 (0 Ce 4f-2)  0.054613 (0 Ce 4f+3)
 MO 29 ::   0.591776 (0 Ce 4f+0)  0.277370 (0 Ce 4f+2)  0.070603 (0 Ce 4f-2)
 MO 30 ::   0.423292 (0 Ce 4f-1)  0.256380 (0 Ce 4f+0)  0.181983 (0 Ce 4f-2)
 MO 31 ::   0.620229 (0 Ce 4f-2)  0.107280 (0 Ce 4f-3)  0.096016 (0 Ce 4f-1)
 MO 32 ::   0.452910 (0 Ce 4f+2)  0.333622 (0 Ce 4f-1)  0.145846 (0 Ce 4f+0)
 MO 33 ::   0.529855 (0 Ce 4f-3)  0.297687 (0 Ce 4f+3)  0.110130 (0 Ce 4f+2)
 MO 34 ::   0.562131 (0 Ce 4f+3)  0.284075 (0 Ce 4f-3)  0.103949 (0 Ce 4f-1)
 MO 35 ::   0.892396 (0 Ce 5dz^2)  0.103652 (0 Ce 5dyz)  0.015317 (0 Ce 5dxy)
 MO 36 ::   0.730298 (0 Ce 5dxz)  0.211169 (0 Ce 5dyz)  0.061383 (0 Ce 5dz^2)
 MO 37 ::   0.684269 (0 Ce 5dyz)  0.261389 (0 Ce 5dxz)  0.059847 (0 Ce 5dz^2)
 MO 38 ::   0.862548 (0 Ce 5dx2-y2)  0.142017 (0 Ce 5dxy)  0.034181 (0 Ce 5dyz)
 MO 39 ::   0.843740 (0 Ce 5dxy)  0.131586 (0 Ce 5dx2-y2)  0.032110 (0 Ce 5dxz)

 CASSCF(1, 7):

CASSCF energy = -8855.07945800899
CASCI E = -8855.07945800899  E(CI) = -1.38604789901365  S^2 = 0.7500000
CASCI state-averaged energy = -8855.079458009
CASCI energy for each state
  State 0 weight 0.142857  E = -8855.07945801303 S^2 = 0.7500000
  State 1 weight 0.142857  E = -8855.07945801302 S^2 = 0.7500000
  State 2 weight 0.142857  E = -8855.0794580091 S^2 = 0.7500000
  State 3 weight 0.142857  E = -8855.07945800898 S^2 = 0.7500000
  State 4 weight 0.142857  E = -8855.07945800702 S^2 = 0.7500000
  State 5 weight 0.142857  E = -8855.07945800613 S^2 = 0.7500000
  State 6 weight 0.142857  E = -8855.07945800568 S^2 = 0.7500000
(1, 0) [-8855.07945801 -8855.07945801 -8855.07945801 -8855.07945801
 -8855.07945801 -8855.07945801 -8855.07945801]
(145, 145)

 New active orbitals:

 MO 20 ::   0.628020 (0 Ce 4dyz)  0.281177 (0 Ce 4dxz)  0.049565 (0 Ce 4dx2-y2)
 MO 21 ::   0.654783 (0 Ce 4dxz)  0.224430 (0 Ce 4dyz)  0.077407 (0 Ce 4dz^2)
 MO 22 ::   0.875862 (0 Ce 4dz^2)  0.097471 (0 Ce 4dyz)  0.021278 (0 Ce 4dxz)
 MO 23 ::   0.878744 (0 Ce 5s)  0.113612 (0 Ce 6s)  0.009934 (0 Ce 4s)
 MO 24 ::   0.748462 (0 Ce 5py)  0.231384 (0 Ce 5px)  0.018487 (0 Ce 5pz)
 MO 25 ::   0.760751 (0 Ce 5px)  0.210978 (0 Ce 5py)  0.026602 (0 Ce 5pz)
 MO 26 ::   0.953243 (0 Ce 5pz)  0.038892 (0 Ce 5py)  0.011632 (0 Ce 7pz)
 MO 27 ::   0.841270 (0 Ce 4f+1)  0.075532 (0 Ce 4f-2)  0.053944 (0 Ce 4f+3)
 MO 28 ::   0.584528 (0 Ce 4f+0)  0.273973 (0 Ce 4f+2)  0.069739 (0 Ce 4f-2)
 MO 29 ::   0.418108 (0 Ce 4f-1)  0.253240 (0 Ce 4f+0)  0.179755 (0 Ce 4f-2)
 MO 30 ::   0.612632 (0 Ce 4f-2)  0.105966 (0 Ce 4f-3)  0.094840 (0 Ce 4f-1)
 MO 31 ::   0.447362 (0 Ce 4f+2)  0.329535 (0 Ce 4f-1)  0.144059 (0 Ce 4f+0)
 MO 32 ::   0.523366 (0 Ce 4f-3)  0.294041 (0 Ce 4f+3)  0.108781 (0 Ce 4f+2)
 MO 33 ::   0.555246 (0 Ce 4f+3)  0.280595 (0 Ce 4f-3)  0.102675 (0 Ce 4f-1)
 MO 34 ::   1.055284 (0 Ce 5dx2-y2)  0.044811 (0 Ce 5dyz)  0.008863 (0 Ce 9dx2-y2)
 MO 35 ::   1.048860 (0 Ce 5dxy)  0.044663 (0 Ce 5dxz)  0.008809 (0 Ce 9dxy)
 MO 36 ::   0.754182 (0 Ce 5dyz)  0.251039 (0 Ce 5dxz)  0.055886 (0 Ce 5dz^2)
 MO 37 ::   0.777569 (0 Ce 5dxz)  0.181854 (0 Ce 5dyz)  0.091730 (0 Ce 5dz^2)
 MO 38 ::   0.951417 (0 Ce 5dz^2)  0.121304 (0 Ce 5dyz)  0.028849 (0 Ce 5dxz)
 MO 39 ::   0.810847 (0 Ce 6s)  0.112775 (0 Ce 5s)  0.094964 (0 Ce 7s)

Generating Spin-Orbit Integrals:

0.02897414391231765

Complex SO-Hamiltonian matrix elements over spin components of spin-free eigenstates:
(In cm-1. Print threshold:    29.000 cm-1)

  I1 =    1 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im = -1015.247
  I1 =    2 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =    68.621
  I1 =    2 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    1 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =   -65.447
  I1 =    3 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =    64.804
  I1 =    3 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    1 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =    61.218
  I1 =    3 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =  -668.421
  I1 =    4 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =    35.601
  I1 =    4 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =   108.519
  I1 =    4 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =    74.602
  I1 =    5 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =   -62.185
  I1 =    5 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =    84.040
  I1 =    5 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    4 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =  -339.648
  I1 =    6 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =   116.443
  I1 =    6 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    4 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =   123.111
  I1 =    6 (E1 =  -8855.07945801) S1 =  0.5 MS1 = -0.5 I2 =    5 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =     0.000 Im =   126.552
  I1 =    7 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    1 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -206.451 Im =   -81.157
  I1 =    7 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -372.528 Im =   135.283
  I1 =    7 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -155.599 Im =  -414.278
  I1 =    7 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    6 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   -30.956 Im =     9.080
  I1 =    8 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   206.451 Im =    81.157
  I1 =    8 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   157.000 Im =   418.276
  I1 =    8 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -369.901 Im =   133.218
  I1 =    8 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    6 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =    10.670 Im =    33.654
  I1 =    8 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    7 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =  1015.247
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   372.528 Im =  -135.283
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    1 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -157.000 Im =  -418.276
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -131.434 Im =   -55.337
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    4 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -296.383 Im =  -584.699
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    5 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   361.613 Im =  -187.677
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    7 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   -68.621
  I1 =    9 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    8 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =    65.447
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   155.599 Im =   414.278
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    1 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   369.901 Im =  -133.218
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   131.434 Im =    55.337
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    4 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -501.171 Im =   259.795
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    5 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -238.538 Im =  -465.088
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    7 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   -64.804
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    8 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   -61.218
  I1 =   10 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    9 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   668.421
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   296.383 Im =   584.699
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   501.171 Im =  -259.795
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    5 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   -72.956 Im =   -26.807
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    6 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -350.353 Im =  -646.342
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    7 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   -35.601
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    9 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =  -108.519
  I1 =   11 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =   10 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   -74.602
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    2 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -361.613 Im =   187.677
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    3 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   238.538 Im =   465.088
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    4 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =    72.956 Im =    26.807
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    6 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =  -802.495 Im =   460.334
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    9 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =    62.185
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =   10 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   -84.040
  I1 =   12 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =   11 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =   339.648
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    0 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =    30.956 Im =    -9.080
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    1 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   -10.670 Im =   -33.654
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    4 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   350.353 Im =   646.342
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =    5 (E2 =  -8855.07945801) S2 =  0.5 MS2 = -0.5 Re =   802.495 Im =  -460.334
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =   10 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =  -116.443
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =   11 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =  -123.111
  I1 =   13 (E1 =  -8855.07945801) S1 =  0.5 MS1 =  0.5 I2 =   12 (E2 =  -8855.07945801) S2 =  0.5 MS2 =  0.5 Re =     0.000 Im =  -126.552

Total energies including SO-coupling:

  State    0 Total energy:  -8855.08578069 | largest |coeff|**2   0.428299 from I =    1 E =  -8855.07945801 S =  0.5
  State    1 Total energy:  -8855.08578069 | largest |coeff|**2   0.428300 from I =    1 E =  -8855.07945801 S =  0.5
  State    2 Total energy:  -8855.08578068 | largest |coeff|**2   0.354279 from I =    2 E =  -8855.07945801 S =  0.5
  State    3 Total energy:  -8855.08578068 | largest |coeff|**2   0.354268 from I =    2 E =  -8855.07945801 S =  0.5
  State    4 Total energy:  -8855.08578068 | largest |coeff|**2   0.428017 from I =    6 E =  -8855.07945801 S =  0.5
  State    5 Total energy:  -8855.08578068 | largest |coeff|**2   0.428010 from I =    6 E =  -8855.07945801 S =  0.5
  State    6 Total energy:  -8855.07471601 | largest |coeff|**2   0.499756 from I =    0 E =  -8855.07945801 S =  0.5
  State    7 Total energy:  -8855.07471601 | largest |coeff|**2   0.499714 from I =    0 E =  -8855.07945801 S =  0.5
  State    8 Total energy:  -8855.07471600 | largest |coeff|**2   0.439043 from I =    2 E =  -8855.07945801 S =  0.5
  State    9 Total energy:  -8855.07471600 | largest |coeff|**2   0.439358 from I =    2 E =  -8855.07945801 S =  0.5
  State   10 Total energy:  -8855.07471600 | largest |coeff|**2   0.500983 from I =    4 E =  -8855.07945801 S =  0.5
  State   11 Total energy:  -8855.07471600 | largest |coeff|**2   0.500687 from I =    4 E =  -8855.07945801 S =  0.5
  State   12 Total energy:  -8855.07471600 | largest |coeff|**2   0.543172 from I =    6 E =  -8855.07945801 S =  0.5
  State   13 Total energy:  -8855.07471600 | largest |coeff|**2   0.543379 from I =    6 E =  -8855.07945801 S =  0.5
[-8855.08578069 -8855.08578069 -8855.08578068 -8855.08578068
 -8855.08578068 -8855.08578068 -8855.07471601 -8855.07471601
 -8855.074716   -8855.074716   -8855.074716   -8855.074716
 -8855.074716   -8855.074716  ]

E 2F(5/2)         = -240959.1927 eV
E 2F(7/2)         = -240958.8916 eV
2F(7/2) - 2F(5/2) =     0.3011 eV
2F(7/2) - 2F(5/2) =  2428.4226 cm-1
